crates/iceberg/src/scan.rs

```diff
@@ -1019,7 +1022,7 @@ mod tests {
                 .metadata(table_metadata)
                 .identifier(TableIdent::from_strs(["db", "table1"]).unwrap())
                 .file_io(file_io.clone())
-                .metadata_location(table_metadata1_location.as_os_str().to_str().unwrap())
+                .metadata_location(template_json_location)
```

Comment:
Before this change, the location in the metadata log and the current location would both be `../metadata/v1.json`.

I wanted to have a distinction so we can assert that `metadata_log_entries` includes the current metadata location, even if not in the metadata log.